src/nft/hooks/useTokenDetails.ts

```diff
@@ -0,0 +1,29 @@
+import { type UseQueryResult, useQuery } from '@tanstack/react-query';
+import { getTokenDetails } from '../../api/getTokenDetails';
+import type { GetTokenDetailsParams, TokenDetails } from '../../api/types';
+import { isNFTError } from '../utils/isNFTError';
+
+export function useTokenDetails({
+  contractAddress,
+  tokenId,
+}: GetTokenDetailsParams): UseQueryResult<TokenDetails> {
+  const actionKey = `useTokenDetails-${contractAddress}-${tokenId}`;
+  return useQuery({
+    queryKey: ['useTokenDetails', actionKey],
+    queryFn: async () => {
+      const tokenDetails = await getTokenDetails({
+        contractAddress,
+        tokenId,
+      });
+
+      if (isNFTError(tokenDetails)) {
+        throw tokenDetails;
+      }
+
+      return tokenDetails;
+    },
+    enabled: true,
```

Comment:
same as prev comment, is there a case where this should be disabled?

Reply:
nope, updated